### PR TITLE
SMF pkg + AMF blueprint pkg

### PR DESCRIPTION
**Pull request by henderiw**
Discussed and reviewed with @s3wong @n2vo @rravindran123 @vjayaramrh viresh (capgemini)

This PR proposes the SMF/AMG BP packages as a reference.

### CODE REVIEW

File: pkg-example-amf-bp/3gpp.yaml
Comment (johnbelamaric) [inline pkg-example-amf-bp/3gpp.yaml:1]: Should this file be named "amf-config.yaml" instead? This name seems a little odd.

I was thinking at some point we will have a 3gpp CRD and may include that CR as a static injection across clusters. When we do that we may want to call that 3gpp.yaml

File: pkg-example-amf-bp/dnn.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: req.nephio.org/v1alpha1
+kind: DataNetworkName
```
Comment (johnbelamaric) [inline pkg-example-amf-bp/dnn.yaml:2]: Didn't we change this to `DataNetwork`?

File: pkg-example-smf-bp/3gpp.yaml
Comment (johnbelamaric) [inline pkg-example-smf-bp/3gpp.yaml:1]: Same comment on naming

File: pkg-example-smf-bp/3gpp.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        - nsmf-event-exposure
+        - nsmf-oam
+
+      sbi:
+        scheme: http
+        registerIPv4: smf-nsmf # IP used to register to NRF
+        bindingIPv4: 0.0.0.0  # IP used to bind the service
+        port: 80
+        tls:
+          key: config/TLS/smf.key
```
Comment (johnbelamaric) [inline pkg-example-smf-bp/3gpp.yaml:24]: Are these files? Where do they come from? I don't see them in the package. Presumably the workload cluster operator (which is generating the actual Deployment), will mount these from somewhere?

File: pkg-example-smf-bp/3gpp.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        addr: 10.100.50.244
+      smfName: SMF
+      snssaiInfos:
+        - sNssai:
+            sst: 1
+            sd: 010203
+          dnnInfos: # DNN information list
+            - dnn: internet # Data Network Name
+              dns: # the IP address of DNS
+                ipv4: 8.8.8.8
```
Comment (johnbelamaric) [inline pkg-example-smf-bp/3gpp.yaml:38]: shouldn't this come from an IPAllocation or some other config? I guess that's getting into "workload configuration" at that point. We haven't figured out how to generate this file from KRM yet.

File: pkg-example-smf-bp/dnn.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+apiVersion: req.nephio.org/v1alpha1
+kind: DataNetworkName
```
Comment (johnbelamaric) [inline pkg-example-smf-bp/dnn.yaml:2]: DataNetwork?

File: pkg-example-upf-bp/dnn.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+apiVersion: req.nephio.org/v1alpha1
+kind: DataNetworkName
```
Comment (johnbelamaric) [inline pkg-example-upf-bp/dnn.yaml:2]: DataNetwork

**Review by johnbelamaric — CHANGES_REQUESTED** [review]

File: pkg-example-amf-bp/3gpp.yaml
Comment (henderiw) [reply pkg-example-amf-bp/3gpp.yaml:1]: we discussed a bit today in a call with multiple people. We need a vendor specific CRD which exposes some information to the workload cluster. None of this is acted upon in specialization e.g., but the info need to be conveyed. So we just added the final config in here. The challenge so far is to find a subset of parameters that are specific to free5gc which need to be interpreted and so far this is void, just for free5gc. So we decided for now to dump this info in a configmap.

Comment (henderiw) [reply pkg-example-amf-bp/3gpp.yaml:1]: updated the name to amf-config

File: pkg-example-smf-bp/3gpp.yaml
Comment (henderiw) [reply pkg-example-smf-bp/3gpp.yaml:1]: updated name to sms-config, did not change the content

File: pkg-example-smf-bp/3gpp.yaml (hunk 2 above)
Comment (henderiw) [reply pkg-example-smf-bp/3gpp.yaml]: This is coming done by the SMF operator. Either we create a specific configmap for SMF/AMF which need to be interpreted bu workload cluster. So this is a vendor specific CRD somehow. We use configmap for now.

**Comment by johnbelamaric** [comment]
/approve

So, I am assuming that for now, we are ignoring stuff in the ConfigMaps, and that we will figure that out later.

File: pkg-example-smf-bp/3gpp.yaml (hunk 3 above)
Comment (henderiw) [reply pkg-example-smf-bp/3gpp.yaml]: DNN we have a reference for. the DNS name was not yet allocated, but we could create a reference to a DNS that was assigned. So far this is the config that SMF needs. So a number of parameters haven't been exposed. However DNS is easy.

File: pkg-example-smf-bp/dnn.yaml (hunk 4 above)
Comment (henderiw) [reply pkg-example-smf-bp/dnn.yaml]: fixed

File: pkg-example-upf-bp/dnn.yaml (hunk 5 above)
Comment (henderiw) [reply pkg-example-upf-bp/dnn.yaml]: fixed.

**Comment by johnbelamaric** [comment]
tagged, all done

File: pkg-example-amf-bp/Kptfile [hunk 6]
```diff
@@ -0,0 +1,13 @@
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: pkg-smf
```
Comment (Smitholi67) [inline pkg-example-amf-bp/Kptfile:4]: name should be pkg-amf right?

Comment (henderiw) [reply pkg-example-amf-bp/Kptfile:4]: fixed

File: pkg-example-amf-bp/Kptfile [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  annotations:
+    config.kubernetes.io/local-config: "true"
+info:
+  description: example smf blueprint package
```
[the remark this answers is not in the document]
Comment (henderiw) [reply pkg-example-amf-bp/Kptfile:8]: fixed

**Pull request merged**